File: cli/examples/indicator/indicator.js
import * as qp from "qpace/node";
import * as pine from "__QPC_PYTHON_PACKAGE__/node";

const ohlcv = qp.Ohlcv.readCsv("btc.csv", qp.Timeframe.Days(1));
const ctx = qp.Ctx(ohlcv, qp.Sym.BTC_USD());

res = pine.indicator.main(ctx.fork(), "close", 30, 14);
console.log(res["locals"]["ma"].slice(0, 90));

File: cli/examples/indicator/indicator.py
import qpace as qp
import __QPC_PYTHON_PACKAGE__ as pine

ohlcv_path = "btc.csv"
ohlcv = qp.Ohlcv.read_csv(ohlcv_path)
ohlcv.timeframe = qp.Timeframe.Days(1)
sym = qp.Sym.BTC_USD()
ctx = qp.Ctx(ohlcv, sym)

res = pine.indicator.main(ctx.copy(), src="close", slow_length=30, fast_length=14)
print(res["locals"]["ma"][0:90])
